src/main/resources/application-development.yml

```diff
@@ -0,0 +1,19 @@
+spring:
+  config:
+    activate:
+      on-profile: development
+
+application:
+  settings:
+    openWeatherMap:
+      api:
+        key: 675bf40b4d52b5fb5e47199ec8315931
+        weatherUrl: https://api.openweathermap.org/data/2.5/weather
+        forecastUrl: https://api.openweathermap.org/data/2.5/forecast
+    yandexMap:
+      api:
+        key:
+        url:
+    restTemplateProperties:
+      connectionTimeout: 3000
+      readTimeout: 3000
```

Comment:
```suggestion
openWeatherApi:
  key: 675bf40b4d52b5fb5e47199ec8315931
  weatherUrl: https://api.openweathermap.org/data/2.5/weather
  forecastUrl: https://api.openweathermap.org/data/2.5/forecast
yandexApi:
  key:
  url:
restTemplateProperties:
  connectionTimeout: 3000
  readTimeout: 3000
```

Reply:
I don't see any point in complicating the structure. The more nested parameters a structure contains, the more difficult it is to implement; you must try to ensure that all parameters are at the first nesting level